extensions/parquet/base/src/main/java/io/deephaven/parquet/base/util/SeekableChannelsProvider.java

```diff
@@ -3,20 +3,51 @@
  */
 package io.deephaven.parquet.base.util;
 
+import io.deephaven.UncheckedDeephavenException;
+import io.deephaven.util.SafeCloseable;
+import io.deephaven.util.annotations.FinalDefault;
 import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.channels.SeekableByteChannel;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
 public interface SeekableChannelsProvider {
 
-    default SeekableByteChannel getReadChannel(@NotNull final String path) throws IOException {
-        return getReadChannel(Paths.get(path));
+    interface ChannelContext extends SafeCloseable {
+
+        ChannelContext NULL = new ChannelContext() {};
+
+        /**
+         * Release any resources associated with this context. The context should not be used afterward.
+         */
+        default void close() {}
+    }
+
+    ChannelContext makeContext();
+
+    interface ContextHolder {
+        void setContext(ChannelContext context);
+
+        @FinalDefault
+        default void clearContext() {
+            setContext(null);
+        }
+    }
+
+    default SeekableByteChannel getReadChannel(@NotNull final ChannelContext context, @NotNull final String uriStr)
+            throws IOException {
+        try {
+            return getReadChannel(context, new URI(uriStr));
+        } catch (final URISyntaxException e) {
+            throw new UncheckedDeephavenException("Cannot convert path string to URI: " + uriStr, e);
+        }
     }
 
-    SeekableByteChannel getReadChannel(@NotNull Path path) throws IOException;
```

Comment:
This is a breaking change; I'm not sure we need to remove it?

Reply:
Since we are reading from URIs only, I thought we can remove the Path API.
And yes, this is breaking, but we are already breaking the interface for passing the context object.
I can confirm with Ryan about this.